classes/Pack.php

```diff
@@ -298,20 +299,30 @@ public static function getQuantity(
             throw new PrestaShopException('Unknown pack stock type');
         }
 
-        // If no pack stock or shop default, set it
-        if (empty($packStockType)
-            || $packStockType == self::STOCK_TYPE_DEFAULT
-        ) {
+        /*
+         * Now, we have resolved how we will calculate the stock of this pack. It can be one of the following.
+         *
+         * STOCK_TYPE_PACK_ONLY	- pack 1pcs + product A 10pcs + product B 20pcs = 1pcs
+         * STOCK_TYPE_PRODUCTS_ONLY - pack 1pcs + product A 10pcs + product B 20pcs = 10 pcs
+         * STOCK_TYPE_PACK_BOTH - pack 1pcs + product A 10pcs + product B 20pcs = 1 pcs
+         */
+
+        // If no pack stock or shop default, set it from configuration
+        if (empty($packStockType) || $packStockType == self::STOCK_TYPE_DEFAULT) {
             $packStockType = Configuration::get('PS_PACK_STOCK_TYPE');
         }
 
-        // Initialize with pack quantity if not only products
+        // If the quantity of the pack depends only on the pack or both packs and products,
+        // we need to load the quantity of the pack from stock_available table.
         if (in_array($packStockType, [self::STOCK_TYPE_PACK_ONLY, self::STOCK_TYPE_PACK_BOTH])) {
-            $packQuantity = $packQuantityInStock;
+            $packQuantity = StockAvailable::getQuantityAvailableByProduct(
```

Comment:
FYI -  I moved this call because the variable was only used in this single condition. It will save us a DB query for packs with `STOCK_TYPE_PRODUCTS_ONLY`.